cpp/src/barretenberg/ecc/scalar_multiplication/pippenger.cpp

```diff
@@ -0,0 +1,51 @@
+#include "pippenger.hpp"
+#include "barretenberg/srs/io.hpp"
+namespace barretenberg {
+namespace scalar_multiplication {
+
+template <typename Curve>
+Pippenger<Curve>::Pippenger(typename Curve::AffineElement* points, size_t num_points)
+    : monomials_(points)
+    , num_points_(num_points)
+{
+    srs::IO<Curve>::byteswap(&monomials_[0], num_points * 64);
+    scalar_multiplication::generate_pippenger_point_table<Curve>(monomials_, monomials_, num_points);
+}
+
+template <typename Curve>
+Pippenger<Curve>::Pippenger(uint8_t const* points, size_t num_points)
+    : num_points_(num_points)
+{
+    monomials_ = point_table_alloc<AffineElement>(num_points);
+
+    srs::IO<Curve>::read_affine_elements_from_buffer(&monomials_[0], (char*)points, num_points * 64);
+    scalar_multiplication::generate_pippenger_point_table<Curve>(monomials_, monomials_, num_points);
+}
+
+template <typename Curve>
+Pippenger<Curve>::Pippenger(std::string const& path, size_t num_points)
+    : num_points_(num_points)
+{
+    monomials_ = point_table_alloc<AffineElement>(num_points);
+
+    srs::IO<Curve>::read_transcript_g1(monomials_, num_points, path);
+    scalar_multiplication::generate_pippenger_point_table<Curve>(monomials_, monomials_, num_points);
+}
+
+template <typename Curve>
+typename Curve::Element Pippenger<Curve>::pippenger_unsafe(ScalarField* scalars, size_t from, size_t range)
```

Comment:
This function and file in general could definitely use some comments. Especially around:
1) The expected scalar/point structure
2) What an "unsafe" pippenger means. That's the first question every newcomer asks when they start working with pippenger in our codebase